src/platform/unittest/TlsTest.cpp

```diff
@@ -19,6 +21,8 @@ const uint32_t DefaultFragmentSize = 1200;
 
 const uint8_t Alpn[] = { 1, 'A' };
 const uint8_t MultiAlpn[] = { 1, 'C', 1, 'A', 1, 'B' };
+const char* TestPath = PfxTestPath;
+const char* PfxPass = "PLACEHOLDER";   // approved for cred scan
```

Comment:
We don't want to check in any binary files. We prefer to generate them dynamically as a part of the prepare-machine.ps1 script.

Reply:
I can do that. What directory this should go to? I assume the prepare-machine.ps1 is called once so it should persist cleaning, right?

Reply:
I don't particularly care where it goes. @anrossi or @thadhouse any suggestions?

Reply:
As I was looking at this agin, it will bring dependency on 'openssl' executable available at time when prepare-machine.ps1 runs. Other option I can possibly see is to build it into the test.ps1 and use OpenSSL we built. It feels like it should not go under build or artifacts as that could be lost during cleanup and one would need to run the  prepare-machine.ps1 again, right? 
I'll wait for  @anrossi or @ThadHouse. 
.NET has similar dislike for binaries so as I mentioned we use base64 in some places.

Reply:
`Prepare-machine.ps1` has a case for windows and for linux.  There's example code in the windows portion for how to generate a certificate+private key and to create a PFX from that certificate which contains the private key. 
Only on linux would you need to call openssl binary to generate the pfx.

Reply:
One more thing, if you put the PFX into a well-known location like `$home` or `$env:userprofile` then it should survive any cleanup.

Reply:
Are you referring to this @anrossi ?
```
.PARAMETER TestCertificates
    Generate test certificates. Only supported for Windows test configuration.
```

There are two issues IMHO: 
How to generate the file and where should it live.

I did consider calling openssl directly but that may be add new dependency and it may be problematic on Windows IMHO. To solve this, I added new script to construct the certificate from cryptographic primitives. That gives us more control (for some parts openssl needs conjuration file and cannot be done via command line arguments) long run and works consistently on all platforms (as New-SelfSignedCertificate works only on Windows)

As far as the location,  `prepare-machine.ps1` runs before the build and it seems to modify global machine state.
Where do you suggest to put the generated pfx file?

Reply:
> One more thing, if you put the PFX into a well-known location like `$home` or `$env:userprofile` then it should survive any cleanup.

I can try that if we are ok that the file lives outside of the build tree. I don't know how much write access CI machine have.

Reply:
I would actually prefer keeping it in the tree. I think it's fine where it is for now.